### ISSUE TITLE
Couldn't get geo data for PT hop

### ISSUE THREAD

**Issue by gfr10598** [m-lab/etl#284]
This message is showing up in the daily NDT pipeline dashboard.

Should add a similar graph in the PT dashboard, and investigate whether there is a major problem.

![image](https://user-images.githubusercontent.com/20915834/31895599-9be93128-b7df-11e7-9ca1-d57f203e775a.png)

**Comment by yachang** [comment]
A quick debug shows that the following IPs did not get any annotation:

A  2017/10/30 18:12:26 geo_annotation.go:103: 172.25.84.137 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 10.150.0.5 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 10.113.0.5 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 10.150.0.5 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.33.254 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.33.254 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.32.158 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:103: 172.17.13.233 cannot get Geo annotation.
 
A  2017/10/30 18:12:26 geo_annotation.go:95: 172.17.10.66 cannot get Geo annotation.
 
From the annotator side, there are those error messages:

A  2017/10/30 18:14:05 handler.go:190: Node not found
 
A  
 
A  2017/10/30 18:14:05 handler.go:190: Node not found
 
A  
 
A  2017/10/30 18:14:05 handler.go:190: Node not found
 
A  
 
A  2017/10/30 18:14:05 handler.go:190: Node not found
 
A  
 
A  2017/10/30 18:14:05 handler.go:190: Node not found
 
A  
 
A  2017/10/30 18:14:06 handler.go:190: Node not found
 
A  
 
Track back the code, the error is from 

https://github.com/m-lab/annotation-service/blob/master/handler/handler.go#L182

node, err = search.SearchBinary(CurrentGeoDataset.IP4Nodes, request.IP)

**Comment by yachang** [comment]
Dig out the PT test name:

10.14.18.42 cannot get Geo annotation 2017/10/24/mlab2.lhr02/20171024T02:01:29Z-181.49.211.34-55418-80.239.170.224-80.paris

**Comment by yachang** [comment]
traceroute [(80.239.170.217:33458) -> (181.49.211.34:55418)], protocol icmp, algo exhaustive, duration 14 s
 1  P(6, 6) 80-239-170-193.customer.teliacarrier.com (80.239.170.193)  0.648/0.779/1.357/0.259 ms 
 2  P(6, 6) ldn-bb3-link.telia.net (62.115.114.234)  0.252/0.901/4.046/1.407 ms 
 3  P(6, 6) nyk-bb3-link.telia.net (62.115.135.94)  70.023/82.912/114.996/17.849 ms 
   MPLS Label 369514 TTL=1
 4  P(6, 6) ash-bb3-link.telia.net (62.115.141.244)  77.138/77.149/77.165/0.009 ms 
 5  P(6, 6) atl-b22-link.telia.net (62.115.137.71)  87.718/89.280/93.567/2.227 ms 
 6  P(6, 6) telmex-ic-316416-atl-b22.c.telia.net (62.115.145.13)  127.911/128.034/128.401/0.173 ms 
 7  P(6, 6) 10.14.18.42 (10.14.18.42)  127.954/131.739/134.528/2.225 ms 
 8  P(0, 6)
 9  P(6, 6) 190.85.241.50 (190.85.241.50)  145.500/253.929/355.379/77.171 ms 
10  P(0, 6)
11  P(6, 6) 181.49.211.34 (181.49.211.34)  142.602/144.079/145.189/0.874 ms 
12  P(6, 6) 181.49.211.34 (181.49.211.34)  143.543/144.641/145.338/0.604 ms 

So the problematic IP is in the middle of hops (step 7). We suspect there is a problem with PT test, OR, some 10.*.*.* IP are owned by some XSP?

**Comment by mlab-code-reviews** [comment]
No, somebody is leaking RFC 1918 addresses (10/8).   By definition they
can't have locations.  We must process this and other undefined locations
sanely.
